Practical Example/RWTakeHome/RWTakeHome/ViewControllers/TutorialViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ViewController.swift
+//  RWTakeHome
+//
+//  Created by Nick Nguyen on 10/7/20.
+//
+
+import UIKit
+
+class TutorialViewController: UIViewController {
+
+  //MARK:- Properties
+
+  let networkService = NetworkService()
+  var tutorials: [Tutorial] = []
+  var videos: [Tutorial] = []
+  var articles: [Tutorial] = []
+  private let cache = Cache<String, Data>()
+  private let photoFetchQueue = OperationQueue()
+  private var operations = [String: Operation]()
+
+  @IBOutlet weak var collectionView: UICollectionView! {
+    didSet { collectionView.dataSource = self }
+  }
+
+  //MARK:- IBActions
+  @IBAction func segmentedControlValueChanged(_ sender: UISegmentedControl) {
+    
+    switch sender.selectedSegmentIndex {
+    case 0: // Article
```

Comment:
One way to make this a little nicer is to define an enum with a raw value.

```swift
enum Section: Int {
  case article = 0
  case video
  case all
}

...

switch Section(rawValue: sender.selectedSegmentIndex) {
  case .video:
    ...
}
```

Reply:
This is really neat. Thank you!